wire-ios-sync-engine/Source/Use cases/UpdateConversationFolderUseCase.swift

```diff
@@ -18,25 +18,21 @@
 
 import WireDataModel
 
-// MARK: - ConversationFolderSelectionUseCaseProtocol
+public struct UpdateConversationFolderUseCase {
 
-public protocol ConversationFolderSelectionUseCaseProtocol {
+    private let context: NSManagedObjectContext
 
-    func invoke<Conversation: ToFolderMovableConversation>(
-        folder: LabelType,
-        conversation: Conversation
-    )
-
-}
-
-// MARK: - ConversationFolderSelectionUseCase
+    public init(context: NSManagedObjectContext) {
+        self.context = context
+    }
 
-final class ConversationFolderSelectionUseCase: ConversationFolderSelectionUseCaseProtocol {
+    public func invoke(conversationID: UUID, folderID: UUID) async throws {
+        await context.perform {
+            guard let folder = Label.fetch(with: folderID, in: context) else { return }
+            guard let conversation = ZMConversation.fetch(with: conversationID, in: context) else { return }
```

Comment:
Instead of returning silently, throw an error. Otherwise the caller might think it succeeded, even though it did nothing.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/146aa3b5af96c9b85e5506e680498bf9c5d41a16